crates/red_knot_python_semantic/src/types/diagnostic.rs

```diff
@@ -474,6 +475,29 @@ declare_lint! {
     }
 }
 
+declare_lint! {
+    /// ## What it does
+    /// Checks for calls to an overloaded function that do not match any of the overloads.
+    ///
+    /// ## Why is this bad?
+    /// Failing to provide the correct arguments to one of the overloads will raise a `TypeError`
+    /// at runtime.
+    ///
+    /// ## Examples
+    /// ```python
+    /// @overload
+    /// def func(x: int): ...
+    /// @overload
+    /// def func(x: bool): ...
```

Comment:
Maybe we can choose something else as an example here, because `int` is a supertype of `bool`, which would mean that the second overload would never be selected(?).